lib/plugin.rb

```diff
@@ -1,10 +1,68 @@
+require 'cocoapods-core'
+require 'keyring_liberator'
+require 'key_master'
+
 module CocoaPodsKeys
 
-  Pod::HooksManager.register('cocoapods-keys', :post_install) do |context, user_options|
-    require 'installer'
-    require 'preinstaller'
+end
+
+module Pod
+  class Installer
+    include Pod::Podfile::DSL
+
+    alias_method :original_install!, :install!
+
+    def install!
+      require 'preinstaller'
+      user_options = config.podfile.plugins["cocoapods-keys"]
+      CocoaPodsKeys::PreInstaller.new(user_options).setup
+
+      # Add our template podspec (needs to be remote, not local). 
+      config.podfile.pod 'Keys', :git => 'https://github.com/ashfurrow/empty-podspec.git', :commit => 'e6588faecb89632f2616251746a313a2ad6336fa'
```

Comment:
This commit is going to change to a tag – it's just temporary until I can figure out how to replace CocoaPods' cached tag commit (see chatroom).